pytensor/link/numba/dispatch/vectorize_codegen.py

```diff
@@ -239,11 +241,21 @@ def codegen(
             output_core_shapes,
         )
 
+        core_signature = typingctx.resolve_function_type(
```

Comment:
This causes compilation of the inner function, so I moved it to the codegen, otherwise we had to pay the cost, even if we could cache it? Not sure. But even if not, I guess we generally want to lazy compile by default?